src/structure/group.rs

```diff
@@ -10,6 +10,7 @@ use super::{entity::Entity, guid::Guid};
 pub struct Group<T> {
     guid: Guid,
     marker: std::marker::PhantomData<T>,
```

Comment:
I wonder if the `Reader`/`Writer` marker type should be moved from the `Group` and into the `Endpoint`

Reply:
> I'm not sure the value of the `Group`

based on the 'structure' module alone, i'm inclined to agree with you. Perhaps the point of that extra layer of abstraction will become apparent in the 'behaviour' or 'discovery' modules...

then again maybe not. This, like many specs of this sort, was designed by committee. You always end up with strange design choices in that environment!